systemtest/src/main/java/io/strimzi/systemtest/utils/FileUtils.java

```diff
@@ -30,132 +30,73 @@ private FileUtils() { }
                 File file = new File(dir, entry.getName());
-                if (entry.isDirectory()) {
-                    file.mkdirs();
-                } else {
-                    fout = new FileOutputStream(file);
+                try (FileOutputStream fout = new FileOutputStream(file)) {
                     while ((len = zin.read(buffer)) != -1) {
                         fout.write(buffer, 0, len);
                     }
-                    fout.close();
                 }
                 entry = zin.getNextEntry();
             }
         } catch (IOException e) {
             LOGGER.error("IOException {}", e.getMessage());
-        } finally {
-            if (fout != null) {
-                try {
-                    fout.close();
-                } catch (IOException e) {
-                    LOGGER.error("IOException {}", e.getMessage());
-                }
-            }
-            if (zin != null) {
-                try {
-                    zin.close();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                }
-            }
         }
-        return dir;
+        return null;
     }
 
-    public static File downloadYamlAndReplaceNameSpace(String url, String namespace) {
-        InputStream bais = null;
-        BufferedReader br = null;
-        OutputStreamWriter osw = null;
+    public static File downloadYamlAndReplaceNameSpace(String url, String namespace) throws IOException {
+        File yamlFile = File.createTempFile("temp-file", ".yaml");
+
+        try (InputStream bais = (InputStream) URI.create(url).toURL().openConnection().getContent();
+             BufferedReader br = new BufferedReader(new InputStreamReader(bais, StandardCharsets.UTF_8));
+             OutputStreamWriter osw = new OutputStreamWriter(new FileOutputStream(yamlFile), StandardCharsets.UTF_8)){
 
-        try {
-            bais = (InputStream) URI.create(url).toURL().openConnection().getContent();
             StringBuilder sb = new StringBuilder();
-            br = new BufferedReader(new InputStreamReader(bais, StandardCharsets.UTF_8));
+
             String read;
             while ((read = br.readLine()) != null) {
-                sb.append(read + "\n");
+                sb.append(read);
+                sb.append("\n");
             }
-            br.close();
             String yaml = sb.toString();
-            File yamlFile = File.createTempFile("temp-file", ".yaml");
-            osw = new OutputStreamWriter(new FileOutputStream(yamlFile), StandardCharsets.UTF_8);
             yaml = yaml.replaceAll("namespace: .*", "namespace: " + namespace);
             yaml = yaml.replace("securityContext:\n" +
                 "        runAsNonRoot: true\n" +
                 "        runAsUser: 65534", "");
             osw.write(yaml);
-            osw.close();
             return yamlFile;
 
         } catch (IOException e) {
             e.printStackTrace();
```

Comment:
Wrapping into a `RuntimeException` is preferred, because it's much less easily overlooked.